### PR TITLE
feat(notifications): Project Search 

### PR DESCRIPTION
**Pull request by mgaeta**
In order for the new NotificationsSettingsByType page to match spec, we need to add back the search bar. This is pretty straightforward, the `renderSearchInput` function does all the heavy lifting.

One project:
Screen Shot 2021-05-25 at 10 55 48 PM

Many projects:
Screen Shot 2021-05-25 at 11 00 05 PM

### CODE REVIEW

File: static/app/views/settings/account/notifications/parentLabel.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import space from 'app/styles/space';
+import {OrganizationSummary, Project} from 'app/types';
+import {getParentKey} from 'app/views/settings/account/notifications/utils';
+
+type Props = {
+  notificationType: string;
+  parent: OrganizationSummary | Project;
+};
+
+/** TODO(mgaeta): Infer parentKey from parent. */
```
Comment (ceorourke) [inline static/app/views/settings/account/notifications/parentLabel.tsx:14]: did you mean to leave this in?

**Comment by ceorourke** [comment]
This is great! Can you add `IconSearch` and a little bit of spacing below the search bar?

File: static/app/views/settings/account/notifications/parentLabel.tsx (hunk 1 above)
Comment (mgaeta) [reply static/app/views/settings/account/notifications/parentLabel.tsx:14]: Yeah I don't know enough about React today but someday I'll be able to simplify this function.

**Comment by mgaeta** [comment]
I actually don't know how to add the icon or mess with margins, can I make that a separate ticket?

**Comment by ceorourke** [comment]
I _think_ (let's be real I have no idea) it'd be something like this: https://github.com/getsentry/sentry/blob/master/static/app/views/settings/organizationMembers/organizationMembersList.tsx#L152-L177 where you wrap it in a styled div to add the spacing at least. I poked around the product more and we don't actually add the icon to any usages of `renderSearchInput` but maybe it's not too hard.

**Review by ceorourke — APPROVED** [review]

**Pull request merged**